notebooks/concepts/Fitting_multiprocess.ipynb

```diff
@@ -0,0 +1,1742 @@
+    "\n",
+    "It hapens that passing a `Spectrum1D` instance to a multiprocess pool worker, prevents the worker from being called by the pool. The spectrum instance does not have even to participate in the actual execution, just a reference to it, if stored in the worker instance, somehow inhibits the worker from being called by the pool.\n",
+    "\n",
+    "This was tested with the worker object as an instance of a class, as below, and also with the worker being a plain top level function to which the spectrum gets passed at every call. No difference. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Worker class"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A class whose instances are callable. It provides the callable for the `Pool.apply_async` function, and also holds everything necessary to perform the fit over one spaxel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class SpaxelWorker:\n",
+    "    def __init__(self, flux_cube, wave_array, initial_model):\n",
+    "        self.cube = flux_cube\n",
+    "        self.wave = wave_array\n",
+    "        self.model = initial_model\n",
+    "        \n",
+    "    def __call__(self, parameters):\n",
+    "        x = parameters[0]\n",
+    "        y = parameters[1]\n",
+    "\n",
+    "        flux = self.cube[:, x, y] * funit\n",
+    "        sp = Spectrum1D(spectral_axis=self.wave, flux=flux)\n",
+    "        fitted_model = fit_lines(sp, self.model)\n",
+    "    \n",
+    "#         print(x, y, fitted_model.amplitude_0.value)\n",
+    "        \n",
+    "        return (x, y, fitted_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Multiprocess with worker pool"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Set up worker"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# (code repeated here for clarity).\n",
+    "region = SpectralRegion(6500*wunit, 7000*wunit)\n",
+    "spectrum = Spectrum1D(spectral_axis=wave, flux=flux3d[:, CENTER, CENTER] * funit)\n",
+    "i1, i2 = _to_edge_pixel(region.subregions[0], spectrum)\n",
+    "\n",
+    "wave_r = wave[i1:i2]\n",
+    "flux_cube_r = flux3d[i1:i2,:,:]\n",
+    "\n",
+    "# Multiprocessing worker.\n",
+    "worker = SpaxelWorker(flux_cube_r, wave_r, fitted_model_central)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Set up callback to collect results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2-D fit results (model parameters) come with different units associated to each 2-D array. So instead of a cube with fit results, we build a list of 2-D arrays storing `Quantity` instances.\n",
+    "\n",
```

Comment:
I don't see why we can't construct a single model cube, with units attached to each parameter.

Reply:
When we combine Quantity arrays with different units in each, into a higher dimension array, they seem to loose their units attributes and revert back to arrays: